### DISCUSSION TITLE
瑞芯微芯片上不支持硬件

### DISCUSSION THREAD

**Discussion by ICECHN** [Tencent/libpag#2231] [General] [answered]
我们在瑞芯微芯片的机顶盒上使用pag，发现无法使用硬解码，会有相关的报错信息
getConfig(0xef8804e0:rk._decoder.avc, ??(0x7f000062)) ERROR: UnsupportedSetting(0x80001019)

说明：
1）我们在realtek芯片上测试pag是正常的
2）瑞芯微芯片上播放mp4视频都是正常的，包括使用yyeva也是正常的

帮忙看下这个是什么问题

**Comment by kevingpqi123** [comment, marked as answer]
可以看下所有包含 BMP 预合成的 pag 文件都无法硬解，还是部分文件无法硬解？如果部分 pag 文件无法硬解，关注下 BMP 预合成的尺寸信息，尽量设置成 8 的倍数
如果所有的文件都无法硬解，你们可以基于 pag 的源码自己适配下或者使用带 ffavc 的版本使用软解
由于人力有限，暂时没有计划去适配机顶盒
